lib/src/screens/debug/scheme_management/scheme_manager_detail_screen.dart

```diff
@@ -0,0 +1,286 @@
+import '../../../widgets/irma_bottom_bar.dart';
+import '../../../widgets/irma_icon_button.dart';
+import '../../../widgets/irma_repository_provider.dart';
+import '../../../widgets/progress.dart';
+import '../../enrollment/provide_email/provide_email_screen.dart';
+import '../../error/error_screen.dart';
+import '../../pin/yivi_pin_screen.dart';
+import '../util/snackbar.dart';
+
+class SchemeManagerDetailScreen extends StatelessWidget {
+  final String schemeManagerId;
+
+  const SchemeManagerDetailScreen(
+    this.schemeManagerId,
+  );
+
+  Future<String?> _requestPin(BuildContext context, String title, String instruction) async {
+    final repo = IrmaRepositoryProvider.of(context);
+    final navigator = Navigator.of(context);
+
+    final hasLongPin = await repo.preferences.getLongPin().first;
+    final maxPinSize = hasLongPin ? 16 : 5;
+
+    return navigator.push(
+      MaterialPageRoute(
+        builder: (context) => YiviPinScaffold(
+          appBar: IrmaAppBar(title: title, hasBorder: false),
+          body: YiviPinScreen(
+            instruction: instruction,
+            pinBloc: EnterPinStateBloc(maxPinSize),
+            maxPinSize: maxPinSize,
+            onSubmit: (pin) => navigator.pop(pin),
+            listener: (context, state) {
+              if (!hasLongPin && state.pin.length == 5) {
+                navigator.pop(state.toString());
+              }
+            },
+          ),
+        ),
+      ),
+    );
+  }
+
+  Future<void> _verifyPin(BuildContext context, String schemeId) async {
+    final repo = IrmaRepositoryProvider.of(context);
+    final navigator = Navigator.of(context);
+
+    final pin = await _requestPin(
+      context,
+      FlutterI18n.translate(context, 'debug.scheme_management.verify_pin.title'),
+      FlutterI18n.translate(
+        context,
+        'debug.scheme_management.verify_pin.content',
+        translationParams: {
+          'scheme': schemeManagerId,
+        },
+      ),
+    );
+    if (pin == null) return;
+
+    repo.bridgedDispatch(AuthenticateEvent(pin: pin, schemeId: schemeId));
+
+    final event = await repo.getEvents().whereType<AuthenticationEvent>().first;
+
+    if (event is AuthenticationErrorEvent) {
+      navigator.push(MaterialPageRoute(
+        builder: (context) => ErrorScreen(
+          details: event.error.toString(),
+          onTapClose: () => navigator.pop(),
+        ),
+      ));
+    } else {
+      showSnackbar(
+        context,
+        event is AuthenticationFailedEvent
+            ? FlutterI18n.translate(
+                context,
+                'debug.scheme_management.verify_pin.failed',
+                translationParams: {
+                  'remainingAttempts': event.remainingAttempts.toString(),
+                  'blockedDuration': event.blockedDuration.toString(),
+                },
+              )
+            : FlutterI18n.translate(context, 'debug.scheme_management.verify_pin.success'),
+      );
```

Comment:
In this file also unchecked usages of `context` are still in place after an `await`.

Reply:
I found out that ability to even do this has been added quite recently: https://docs.flutter.dev/release/release-notes/release-notes-3.7.0 

> Add BuildContext.mounted by @goderbauer in https://github.com/flutter/flutter/pull/111619

Reply:
Looks like it is indeed required in this case. https://api.flutter.dev/flutter/widgets/BuildContext/mounted.html 

> If a [BuildContext](https://api.flutter.dev/flutter/widgets/BuildContext-class.html) is used across an asynchronous gap (i.e. after performing an asynchronous operation), consider checking [mounted](https://api.flutter.dev/flutter/widgets/BuildContext/mounted.html) to determine whether the context is still valid before interacting with it